set-refs-from-tdx-manifest.py

```diff
@@ -0,0 +1,534 @@
+#!/usr/bin/env python3
+# /// script
+# requires-python = ">=3.10"
+# dependencies = [
+#     "defusedxml",
+# ]
+# ///
+
+"""
+Update the submodules in this repository from a git ref of git.toradex.com/toradex-manifest.git
+
+git.toradex.com/toradex-manifest.git is a google repotool manifest for an openembedded layer set, and is how toradex
+distributes their official versions of their openembedded tree. When they release a new version, they do it by
+pushing a branch there. This tool updates our submodule-style openembedded tree from the manifests there (among other
+things).
+"""
+
+from typing import Iterator, Any, Iterable
+from dataclasses import dataclass
+import logging
+import argparse
+import sys
+import traceback
+import contextlib
+from shutil import rmtree
+import subprocess
+import tempfile
+from itertools import chain
+from pathlib import Path
+from xml.etree.ElementTree import ElementTree
+
+try:
+    from defusedxml import ElementTree as ET
+except ImportError:
+    sys.stderr.write(
+        "This script requires https://pypi.org/project/defusedxml/. Run pip install defusedxml or run this script with uv.\n"
+    )
+    raise
+    sys.exit(-1)
+
+LOG = logging.getLogger()
+
+_MANIFEST_REPO = "http://git.toradex.com/toradex-manifest.git"
+
+_REPO_PATH_OVERRIDES = {"tools/bitbake": "layers/openembedded-core/bitbake"}
+
+
+@dataclass
+class SubmoduleSpec:
+    """Track a git submodule"""
+
+    tracking: str | None
+    sha: str
+    name: str
+    path: str
+    url: str
+
+
+def _spec_from_lines(git_lines: str) -> Iterator[SubmoduleSpec]:
+    """Read a submodule spec from lines in .gitmodules"""
+    for git_line in git_lines:
+        if not git_line:
+            continue
+        flags = git_line[:2]
```

Comment:
what are submodule flags? also seems not used anywhere here?

Reply:
they're little characters for like "modified since checkout". i guess i could drop the variable